### ISSUE TITLE
Bulk user fetch for list of IDs

### ISSUE THREAD

**Issue by david-crespo** [oxidecomputer/omicron#1209]
As a kind of workaround for #1207, it would be helpful to be able to fetch info on a specified set of users. We could put a comma-separated list of IDs in the query string.

I was initially concerned about the query string getting too long (UUID v4 is 36 characters), but as far as I can tell browser limits on query string length are pretty high. It's only Edge that's imposing any kind of limit we'd worry about, and I can't tell from this garbagey website whether it applies to background requests. [Some places](https://www.geeksforgeeks.org/maximum-length-of-a-url-in-different-browsers/) claim it's around 2k chars, [others](https://stackoverflow.com/questions/32267442/url-length-limitation-of-microsoft-edge) claim it's 4k chars. It's worth looking into this further. If it doesn't work in the query string, we could use a POST.

<img width="828" alt="image" src="https://user-images.githubusercontent.com/3612203/173624220-b9131e75-4bde-483c-85d6-0300fd5a3b9e.png">

<img width="745" alt="image" src="https://user-images.githubusercontent.com/3612203/173624657-f2dc826f-4c49-4cfa-a13f-70cf86a710eb.png">

**Comment by davepacheco** [comment]
Is it even worth doing this in bulk at this point or is that premature optimization?

**Comment by david-crespo** [comment]
Well, so far the number of users is almost always less than the max page size of 1000. But that could change soon and suddenly, couldn't it?

**Comment by davepacheco** [comment]
For sure.  I was envisioning an API that fetches the details for one user, identified by id.  I realize this could get bad in pathological cases.  I'm expecting that most resources aren't going to have that many items in the policy, especially if customers are inclined to use groups.

**Comment by david-crespo** [comment]
Yeah, fetching for one user doesn't really help much if we're trying to display the table. I agree that groups are going to be the main way this is done. We also have this problem for group names, though in general there should be a lot fewer of them, so fetching all of them has a much longer lifetime as a temporary solution.

**Comment by davepacheco** [comment]
Can't you use the one-at-a-time API the same way you'd use the bulk API, just making separate requests?

Yeah, I think we'd need an analogous ones for groups.

**Comment by david-crespo** [comment]
You can, but browsers limit the number of concurrent requests to 4-8 ([source](https://stackoverflow.com/a/985704/604986)), so that gets sad fast if you're doing 20. It could be ok for groups, though in that case I'd probably just pull them all in one request.

**Comment by davepacheco** [comment]
I imagine another prerequisite here is having any useful information about Silo Users.  In #1261 they'll only have an id, so this doesn't seem useful.  After #1210 they'll have an `external_id`, but even that may not generally be useful.

**Comment by david-crespo** [comment]
Yes, definitely. That broad problem may be a big deal for role assignment in the console. The UI for picking a user is not usable if you don't have a human-readable display name to choose them by.
